src/main/java/dev/blaauwendraad/masker/json/MaskingState.java

```diff
@@ -104,23 +104,31 @@ public void backtrackCurrentJsonPath() {
         currentJsonPath.pop();
     }
 
-    /**
-     * Checks if the last segment of the current jsonpath is an array index.
-     */
-    public boolean isInArray() {
-        return !currentJsonPath.isEmpty() && currentJsonPath.peek().offset == -1;
-    }
-
     /**
      * Increments an array index in the last segment of the current jsonpath.
      * Throws {@link java.lang.IllegalStateException} if the last segment is not an array index.
      */
     public void incrementCurrentJsonPathArrayIndex() {
```

Comment:
Does this really need to be public?

Reply:
Not really, I can make it package-private.
Ideally I would move all jsonpath-related logic into a separate "JsonPathTracker" class aggregated into MaskingState. But as we are going to start on token-based state machine that would not be necessary right now.